### PR TITLE
Add spelling check

### LINKED ISSUE #1260: Add spellcheck action

### PR DESCRIPTION
**Pull request by cptartur**
Closes #1260

## Introduced changes

- Added spelling check to CI
- Updated relevant documents

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by piotmag769 — APPROVED** [review]
🔥✈️💧

File: _typos.toml
Comment (drknzz) [inline _typos.toml:1]: What dis

File: docs/src/development/environment-setup.md [hunk 1]
```diff
@@ -57,11 +57,27 @@ $ cargo fmt
...
 
 Or using our defined alias
 
 ```shell
 $ cargo lint
 ```
+
+## Spelling
+
+Starknet Foundry uses [typos](https://github.com/marketplace/actions/typos-action) for spelling checks.
```
[the remark this answers is not in the document]
Comment (cptartur) [reply docs/src/development/environment-setup.md:71]: The installation instructions in the GH marketplace are exactly the same as on crates.io and it also indicates what action we are using clearly.

File: _typos.toml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[default.extend-words]
+ba = "ba"
```
[the remark this answers is not in the document]
Comment (cptartur) [reply _typos.toml:2]: Right here https://github.com/foundry-rs/starknet-foundry/blob/f86f403e3d1c9b3a23659473a549c3d0d6507b9f/scripts/snfoundryup#L9 😅

File: _typos.toml
Comment (cptartur) [reply _typos.toml:1]: Configuration of the `typos`, we had some false positive for this line https://github.com/foundry-rs/starknet-foundry/blob/f86f403e3d1c9b3a23659473a549c3d0d6507b9f/scripts/snfoundryup#L9 and also a google token defined somewhere in docs.

**Pull request merged**